compiler/rustc_query_system/src/dep_graph/serialized.rs

```diff
@@ -391,20 +402,55 @@ struct NodeInfo {
 
 impl NodeInfo {
     fn encode<D: Deps>(&self, e: &mut FileEncoder) {
-        let header = SerializedNodeHeader::<D>::new(self);
+        let NodeInfo { node, fingerprint, ref edges } = *self;
+        let header =
+            SerializedNodeHeader::<D>::new(node, fingerprint, edges.max_index(), edges.len());
+        e.write_array(header.bytes);
+
+        if header.len().is_none() {
+            e.emit_usize(edges.len());
+        }
+
+        let bytes_per_index = header.bytes_per_index();
+        for node_index in edges.iter() {
+            e.write_with(|dest| {
+                *dest = node_index.as_u32().to_le_bytes();
+                bytes_per_index
+            });
+        }
+    }
+
+    #[inline]
+    fn encode_promoted<D: Deps>(
```

Comment:
Could you add a comment on this method, to explain the difference with `encode`?